docs/getting_started/tutorial.rst

```diff
@@ -284,143 +398,167 @@ Adjust your blog page template to include the image:
+   :alt: Tagging a post
 
+To render tags on a ``BlogPage``, add this to ``blog_page.html:``
 
-    class BlogIndexPage(Page):
-        intro = RichTextField(blank=True)
+.. code-block:: html+django
 
-        content_panels = Page.content_panels + [
-            FieldPanel('intro', classname="full"),
-            InlinePanel('related_links', label="Related links"),
-        ]
+    {% if page.specific.tags.all.count %}
+        <div class="tags">
+            <h3>Tags</h3>
+            {% for tag in page.specific.tags.all %}
+                <a href="{% slugurl 'tags' %}?tag={{ tag }}"><button type="button">{{ tag }}</button></a>
+            {% endfor %}
+        </div>
+    {% endif %}
+
+Visiting a blog post with tags should now show a set of linked
+buttons at the bottom - one for each tag. However, clicking a button
+will get you a 404, since we haven't yet defined a "tags" view, which
+is going to require a little extra magic. Add to ``models.py:``
 
+.. code-block:: python
 
-    class BlogIndexRelatedLink(Orderable, RelatedLink):
-        page = ParentalKey('BlogIndexPage', related_name='related_links')
+    class BlogTagIndexPage(Page):
 
-.. figure:: ../_static/images/tutorial/tutorial_7.png
-   :alt: Blog index edit screen
+        def get_context(self, request):
+
+            # Filter by tag
+            tag = request.GET.get('tag')
+            blogpages = BlogPage.objects.filter().filter(tags__name=tag)
+
+            # Update template context
+            context = super(BlogTagIndexPage, self).get_context(request)
+            context['blogpages'] = blogpages
+            return context
 
-Extend ``blog_index_page.html`` to show related items
+Note that this Page-based model defines no fields of its own.
+Even without fields, subclassing ``Page`` makes it a part of the
+Wagtail ecosystem, so that you can give it a title and URL in the
+admin, and so that you can manipulate its contents by returning
+a queryset from its ``get_context()`` method.
+
+Migrate this in, then create a new ``BlogTagIndexPage`` in the admin.
+You'll probably want to create the new page/view under Homepage,
+parallel to your Blog index. Give it the slug "tags" on the Promote tab.
+
+Access ``/tags`` and Django will tell you what you probably already knew:
+you need to create a template ``blog/blog_tag_index_page.html:``
 
 .. code-block:: html+django
 
     {% extends "base.html" %}
-
     {% load wagtailcore_tags %}
 
-    {% block body_class %}template-blogindexpage{% endblock %}
-
     {% block content %}
-        <h1>{{ page.title }}</h1>
-
-        <div class="intro">{{ page.intro|richtext }}</div>
 
-        {% if page.related_links.all %}
-            <ul>
-                {% for item in page.related_links.all %}
-                    <li><a href="{{ item.link }}">{{ item.title }}</a></li>
-                {% endfor %}
-            </ul>
+        {% if request.GET.tag|length %}
+            <h4>Showing pages tagged "{{ request.GET.tag|safe }}"</h4>
```

Comment:
We definitely don't want to use `|safe` here - the GET parameter is unchecked user input, and this will create an XSS vulnerability.